File: code-examples/scala/connection/socks.scala
import org.mongodb.scala._

import scala.concurrent.ExecutionContext.Implicits.global
import scala.concurrent.Future

// Replace the placeholder with your connection string
val mongoConnectionString = "<connection string uri>"

// Substitute placeholders with your SOCKS5 proxy server details
val settings = MongoClientSettings.builder()
  .applyToClusterSettings(b => b.hosts(List(new ServerAddress("<host>", 1080)).asJava))
  .credential(MongoCredential.createCredential("<username>", "admin", "<password>".toCharArray))
  .build()

// Create a new client with the proxy server details
val mongoClient: MongoClient = MongoClient(mongoConnectionString, settings)

val database: MongoDatabase = mongoClient.getDatabase("myDB")
val collection: MongoCollection[Document] = database.getCollection("myColl")

def runQuery(): Future[Document] = {
  collection.find().first().head()
}

runQuery().onComplete(println)

mongoClient.close()

File: code-examples/php/connection/socks.php
<?php

require 'vendor/autoload.php';

// Replace the placeholders with your connection string and SOCKS5 proxy server details
$uri = '<connection string uri>';
$options = [
    'driver' => 'mongodb',
    'host' => '<host>',
    'port' => 1080,
    'username' => '<username>',
    'password' => '<password>',
];

$client = new MongoDB\Client($uri, $options);
$collection = $client->selectCollection('myDB', 'myColl');

try {
    // Retrieve the first document from the MongoDB collection and print it
    $document = $collection->findOne();
    print_r($document);
} catch (Exception $e) {
    echo 'Exception: ', $e->getMessage(), "\n";
} finally {
    // Close the connection after the operation completes
    $client = null;
}
